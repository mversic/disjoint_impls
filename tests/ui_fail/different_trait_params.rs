use disjoint_impls::disjoint_impls;

pub trait Dispatch<T> {
    type Group;
}

pub enum GroupA {}
impl Dispatch<u32> for String {
    type Group = GroupA;
}
impl<T> Dispatch<i32> for Vec<T> {
    type Group = GroupA;
}

pub enum GroupB {}
impl Dispatch<i32> for i32 {
    type Group = GroupB;
}
impl Dispatch<u32> for u32 {
    type Group = GroupB;
}

disjoint_impls! {
    pub trait Kita {
        const NAME: &'static str;
    }

    impl<T: Dispatch<i32, Group = GroupA>> Kita for T {
        const NAME: &'static str = "Blanket A";
    }
    impl<T: Dispatch<u32, Group = GroupB>> Kita for T {
        const NAME: &'static str = "Blanket B";
    }
}

fn main() {}

