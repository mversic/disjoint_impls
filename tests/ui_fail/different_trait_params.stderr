error: Conflicting implementations of `impl Kita for _ŠČ0`
  --> tests/ui_fail/different_trait_params.rs:23:1
   |
23 | / disjoint_impls! {
24 | |     pub trait Kita {
25 | |         const NAME: &'static str;
...  |
34 | | }
   | |_^
   |
   = note: this error originates in the macro `disjoint_impls` (in Nightly builds, run with -Z macro-backtrace for more info)
